src/Storages/MessageQueueSink.h

```diff
@@ -33,13 +33,17 @@ class MessageQueueSink : public SinkToStorage
         const String & storage_name_,
         const ContextPtr & context_);
 
+    ~MessageQueueSink() override
+    {
+        onFinish();
```

Comment:
That is not cleat if it could throw an exception. Will revert changes in that class.